### DISCUSSION TITLE
Will it be possible to script games in languages other than Lua?

### DISCUSSION THREAD

**Discussion by Clockwork-Muse** [o3de/o3de#7916] [Ideas]
In my case, it would be nice to be able to script in python (since I'm look at using the engine for machine learning-related scenarios).

If not officially supported, would this be something a separate gem be able to do?

**Comment by southeast2017** [comment]
Mark! I also want to know!!

**Comment by BytesOfPiDev** [comment]
It is possible to add a new scripting language via a gem, yes. ScriptCanvas, O3DE's visual scripting language, is a gem. You would use AZ::BehaviorContext and bind its functions/callbacks to whatever language you're implementing. I've never fiddled with it, but I remember some people talking about it before.
